compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5303,10 +5303,25 @@ private BType checkInvocationParam(BLangInvocation iExpr) {
             return symTable.noType;
         }
 
-        List<BType> paramTypes = ((BInvokableType) iExpr.symbol.type).getParameterTypes();
+        BInvokableSymbol invokableSymbol = ((BInvokableSymbol) iExpr.symbol);
+        List<BType> paramTypes = ((BInvokableType) invokableSymbol.type).getParameterTypes();
 
-        int parameterCount = paramTypes.size();
+        int parameterCountForPositionalArgs = paramTypes.size();
+        int parameterCountForNamedArgs = parameterCountForPositionalArgs + invokableSymbol.includedRecordParams.size();
         iExpr.requiredArgs = new ArrayList<>();
+        for (BVarSymbol symbol : invokableSymbol.params) {
+            if (Symbols.isFlagOn(Flags.asMask(symbol.getFlags()), Flags.INCLUDED)) {
+                LinkedHashMap<String, BField> fields = ((BRecordType) symbol.type).fields;
+                if (!fields.isEmpty()) {
+                    for (String field : fields.keySet()) {
+                        if (fields.get(field).type.tag != TypeTags.NEVER) {
+                            parameterCountForNamedArgs = parameterCountForNamedArgs - 1;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
 
```

Comment:
Shall we rearrange the login inside for loop like this, it will be more readable,

```
for (BVarSymbol symbol : invokableSymbol.params) {
   if (!Symbols.isFlagOn(Flags.asMask(symbol.getFlags()), Flags.INCLUDED)) {
       continue;
   }
   .......
   if (fields.isEmpty()) {
       continue;
   }
   for (String field : fields.keySet()) {
   }
}
```

Reply:
Changed.